src/guest_memory.rs

```diff
@@ -288,6 +288,11 @@ pub trait GuestMemoryRegion: Bytes<MemoryRegionAddress, E = Error> {
     fn as_volatile_slice(&self) -> Result<volatile_memory::VolatileSlice> {
         self.get_slice(MemoryRegionAddress(0), self.len() as usize)
     }
+
+    /// Returns true if the region is backed by hugetlbfs
+    fn is_hugetlbfs(&self) -> bool {
```

Comment:
Hmm, would it actually make sense to return an `Option<bool>`, where `None` represents that no information is available? In our particular situation (where `is_hugetlbfs` has to be set via the `set_hugetlbfs` accesor), this would also be a natural fit since the field starts out as `None`, and only becomes `true` or `false` if it's explicitly set. Just wondering what other folks think about this approach.

Reply:
Updated.